camel-core/src/main/java/org/apache/camel/component/file/GenericFileProducer.java

```diff
@@ -274,7 +274,11 @@ public void writeFile(Exchange exchange, String fileName) throws GenericFileOper
             log.trace("About to write [{}] to [{}] from exchange [{}]", new Object[]{fileName, getEndpoint(), exchange});
         }
 
-        boolean success = operations.storeFile(fileName, exchange);
```

Comment:
i believe you are right. i did not want such change IMHO as well. there reason why i went there was camel-zipfile was passing either ZipIterator or Outputstream to producer. For the case in the issue, iterator usage is required for Splitter EIP and Message is created with FILENAME header and conveyed to producer and core file producer does all the thing like creating directories and and files on the producer endpoint. What i did in camel-zipfile component to create a Message with FILENAME header <path> + "." which only indicated the target empty directory to be created. And core file producer already does manupulation like creating paths and files on producer endpoint. I just added a simple condition to bypass storeFile method call and directory creation for empty folders is OK. that was the purpose. I have not come up with an easier solution  and i did not feel like i would do something dirty on core file producer, therefore, i chose that path. If this does not make any sense, i can close the PR.